libp2p/peer/peerinfo.py

```diff
@@ -1,19 +1,39 @@
-from typing import List, Sequence
+import json
+from typing import Any, List, Sequence
 
 import multiaddr
 
 from .id import ID
 
 
 class PeerInfo:
-
     peer_id: ID
     addrs: List[multiaddr.Multiaddr]
 
     def __init__(self, peer_id: ID, addrs: Sequence[multiaddr.Multiaddr]) -> None:
         self.peer_id = peer_id
         self.addrs = list(addrs)
 
+    def to_string(self) -> str:
+        return json.dumps(
+            [self.peer_id.to_string(), list(map(lambda a: str(a), self.addrs))]
+        )
+
+    def __eq__(self, other: Any) -> bool:
+        return (
+            isinstance(other, PeerInfo)
+            and self.peer_id == other.peer_id
+            and self.addrs == other.addrs
+        )
+
+    @classmethod
+    def info_from_string(cls, info: str) -> "PeerInfo":
```

Comment:
On second thought, would it be more reasonable to move it to `kademlia_peer_router.py` as a standalone function whose input is `PeerInfo`? Probably something like `peer_info_from_str`. It seems `PeerInfo.info_from_string` is only used in `KadmeliaPeerRouter` and probably not in the rest of the codebase if it is just a workaround?